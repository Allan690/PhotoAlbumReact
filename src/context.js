import React from "react";
import axios from "./axiosConfig";

const initialState = { photos: [] };
const store = React.createContext(null);
const { Provider } = store;

const reducer = (state, action) => {
  switch (action.type) {
    case "SET_PHOTOS":
      return {
        ...state,
        photos: action.payload,
      };
    default:
      return new Error();
  }
};

const AppProvider = ({ children }) => {
  const [state, dispatch] = React.useReducer(reducer, initialState);
  const fetchPhotos = async (id = 2) => {
    const result = await axios.request({
      method: "get",
      url: `/${id}/photos?_limit=20`,
    });
    const { data } = result;
    // @ts-ignore
    dispatch({
      type: "SET_PHOTOS",
      payload: data,
    });
  };

  const actions = {
    fetchPhotos,
  };

  return (
    <Provider
      value={{
        state,
        actions,
      }}
    >
      {children}
    </Provider>
  );
};

export function usePhotoState() {
  return React.useContext(store).state;
}

export function usePhotoActions() {
  return React.useContext(store).actions;
}

export { store, AppProvider };
