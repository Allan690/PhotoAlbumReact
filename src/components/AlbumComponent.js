import React from 'react';
import '../assets/css/AlbumComponent.css';
import Photo from './PhotosComponent';
import { usePhotoState } from '../context';

const AlbumComponent = () => {
  const { photos } = usePhotoState();
  return (
     <div className="album">
        { photos.map((photo, ind) => <Photo {...photo } key={ind}/>) }
     </div>
 )
}

export default AlbumComponent;
