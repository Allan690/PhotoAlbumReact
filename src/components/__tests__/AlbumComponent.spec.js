import React from "react";
import AlbumComponent from "../AlbumComponent";
import Photo from "../PhotosComponent";
import { usePhotoState } from "../../context";
import { shallow } from "enzyme";
jest.mock("../../context");

describe("AlbumComponent", () => {
  const photosArray = [
    {
      albumId: 1,
      id: 1,
      title: "accusamus beatae ad facilis cum similique qui sunt",
      url: "https://via.placeholder.com/600/92c952",
      thumbnailUrl: "https://via.placeholder.com/150/92c952",
    },
    {
      albumId: 1,
      id: 2,
      title: "reprehenderit est deserunt velit ipsam",
      url: "https://via.placeholder.com/600/771796",
      thumbnailUrl: "https://via.placeholder.com/150/771796",
    },
    {
      albumId: 1,
      id: 3,
      title: "officia porro iure quia iusto qui ipsa ut modi",
      url: "https://via.placeholder.com/600/24f355",
      thumbnailUrl: "https://via.placeholder.com/150/24f355",
    },
  ];

  beforeEach(() => {
    usePhotoState.mockReturnValue({
        photos: photosArray,
      });
  });

  afterEach(() => jest.resetAllMocks());

  it("renders photo album correctly", () => {
    const wrapper = shallow(<AlbumComponent />);
    expect(wrapper).toMatchSnapshot();
  });

  it('contains album and photos', () => {
    const wrapper = shallow(<AlbumComponent />);
    expect(wrapper.find(".album")).toHaveLength(1);
    expect(wrapper.contains(<Photo {...photosArray[0]} />)).toEqual(true);
    expect(wrapper.contains(<Photo {...photosArray[1]} />)).toEqual(true);
    expect(wrapper.contains(<Photo {...photosArray[2]} />)).toEqual(true);
  });
});
