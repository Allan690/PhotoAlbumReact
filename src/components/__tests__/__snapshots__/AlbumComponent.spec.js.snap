// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumComponent renders photo album correctly 1`] = `
<div
  className="album"
>
  <Photo
    albumId={1}
    id={1}
    key="0"
    thumbnailUrl="https://via.placeholder.com/150/92c952"
    title="accusamus beatae ad facilis cum similique qui sunt"
    url="https://via.placeholder.com/600/92c952"
  />
  <Photo
    albumId={1}
    id={2}
    key="1"
    thumbnailUrl="https://via.placeholder.com/150/771796"
    title="reprehenderit est deserunt velit ipsam"
    url="https://via.placeholder.com/600/771796"
  />
  <Photo
    albumId={1}
    id={3}
    key="2"
    thumbnailUrl="https://via.placeholder.com/150/24f355"
    title="officia porro iure quia iusto qui ipsa ut modi"
    url="https://via.placeholder.com/600/24f355"
  />
</div>
`;
